elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+                                (DeselectOption)
+                             else
+                                (SelectOption name)
+                            )
+                        ]
+                        [ tt [] [ text name ] ]
+                    ]
+                )
+              ]
+            , (if not isSelected then
+                []
+               else
+                [ tr []
+                    [ td [ class "details" ] [ describe_option option ]
+                    ]
+                ]
+              )
+            ]
+        )
+
+
+opt_tuple_matches_name : String -> ( String, Option ) -> Bool
+opt_tuple_matches_name term ( name, option ) =
+    term == name
+
+
+changePageIf : Bool -> Int -> String -> Html Msg
+changePageIf cond page txt =
+    button
+        (if cond then
+            [ onClick (ChangePage page) ]
+         else
+            [ disabled True ]
+        )
+        [ text txt ]
+
+
+view : Model -> Html Msg
+view model =
+    if List.length model.options > 0 then
+        viewOptions model
+    else
+        div [] [ h1 [] [ text (toString model.status) ] ]
+
+
+debounce =
+    Debounce.debounceCmd debounce_cfg
+
+
+debounce_cfg : Debounce.Config Model Msg
+debounce_cfg =
+    Debounce.config
+        .debounce_state
+        -- getState   : Model -> Debounce.State
+        (\model s -> { model | debounce_state = s })
+        -- setState   : Debounce.State -> Model -> Debounce.State
+        Bounce
+        -- msgWrapper : Msg a -> Msg
+        200
+
+
+
+-- timeout ms : Float
+
+
+viewOptions : Model -> Html Msg
+viewOptions model =
+    div []
+        [ p []
+            [ input [ id "search", class "search-query span3", autofocus True, value model.query, onInput ChangeQuery ] []
+            ]
+        , hr [] []
+        , p [ id "how-many" ]
+            [ em []
+                [ text
+                    (if (List.length model.matchingOptions) == 0 then
+                        "Showing no results"
+                     else
+                        (String.concat
+                            [ "Showing results "
+                            , (toString (((model.page - 1) * 15) + 1))
+                            , "-"
+                            , (toString (min (List.length model.matchingOptions) ((model.page) * 15)))
+                            , " of "
+                            , (toString (List.length model.matchingOptions))
+                            , "."
+                            ]
+                        )
+                    )
+                ]
+            ]
+        , table [ class "options table table-hover" ]
+            [ thead []
+                [ tr []
+                    [ th [] [ text "Option name" ]
+                    ]
+                ]
+            , tbody []
+                (List.concat (List.map (optionToTd model.selected) (fetch_page model.page model.matchingOptions)))
+            ]
+        , ul [ class "pager" ]
+            [ li []
```

Comment:
You could potentially format these a bit different (i.e. `li [] [ (changePageIf (not (model.page <= 1)) 1 "« First") ]` could go on a single line, pretty sure `elm-format` would allow it, too :)